src/screen.rs

```diff
@@ -1099,7 +1147,11 @@
         for i in 0..zelf.desired.line_count() {
             zelf.actual.create_line(i);
 
-            let start_pos = if i == 0 { left_prompt_width } else { 0 };
+            let start_pos = if i == 0 && !zelf.scrolled() {
```

Comment:
> This fixes an issue where, when scrolled, the space that the prompt would
 > occupy (if not scrolled) is not rendered.

How to reproduce?
If I run

	fish -C 'bind ctrl-g \'commandline $(seq $LINES)\''

and type `ctrl-g shift-enter` things work as expected.
The (one-line) prompt is only rendered if we're not scrolled.
Maybe this is about multi-line prompts?

Reply:
Right that makes sense.
FWIW we might want to rework scrolling a bit, see https://github.com/fish-shell/fish-shell/issues/11029